# xarm:
# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# Position Controllers ---------------------------------------
joint1_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint1
  pid: {p: 1200.0, i: 5.0, d: 10.0}
joint2_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint2
  pid: {p: 1400.0, i: 5.0, d: 10.0}
joint3_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint3
  pid: {p: 1200.0, i: 5.0, d: 5.0}
joint4_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint4
  pid: {p: 850.0, i: 3.0, d: 5.0}
joint5_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint5
  pid: {p: 500.0, i: 3.0, d: 1.0}
joint6_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint6
  pid: {p: 500.0, i: 1.0, d: 1.0}
joint7_position_controller:
  type: effort_controllers/JointPositionController
  joint: joint7
  pid: {p: 300.0, i: 0.05, d: 1.0}


# Effort Controllers ---------------------------------------
joint1_effort_controller:
  type: effort_controllers/JointEffortController
  joint: joint1
  pid: {p: 1200.0, i: 5.0, d: 10.0}
joint2_effort_controller:
  type: effort_controllers/JointEffortController
  joint: joint2
  pid: {p: 1400.0, i: 5.0, d: 10.0}
joint3_effort_controller:
  type: effort_controllers/JointEffortController
  joint: joint3
  pid: {p: 1200.0, i: 5.0, d: 5.0}
joint4_effort_controller:
  type: effort_controllers/JointEffortController
  joint: joint4
  pid: {p: 850.0, i: 3.0, d: 5.0}
joint5_effort_controller:
  type: effort_controllers/JointEffortController
  joint: joint5
  pid: {p: 500.0, i: 3.0, d: 1.0}
joint6_effort_controller:
  type: effort_controllers/JointEffortController
  joint: joint6
  pid: {p: 500.0, i: 1.0, d: 1.0}
joint7_effort_controller:
  type: effort_controllers/JointEffortController
  joint: joint7
  pid: {p: 300.0, i: 0.05, d: 1.0}

